badgecheck/util.py

```diff
@@ -0,0 +1,48 @@
+import string
+from urlparse import urlparse
+
+from pyld.jsonld import JsonLdError
+import requests
+import requests_cache
+
+
+class CachableDocumentLoader(object):
+    def __init__(self, cachable=False):
+        self.cachable = cachable
+        if self.cachable:
+            self.session = requests_cache.CachedSession(backend='memory', expire_after=300)
+        else:
+            self.session = requests.Session()
+
+    def __call__(self, url):
+        try:
+            # validate URLs
+            pieces = urlparse(url)
+            if (not all([pieces.scheme, pieces.netloc]) or
+                    pieces.scheme not in ['http', 'https'] or
+                    set(pieces.netloc) > set(string.ascii_letters + string.digits + '-.:')):
+                raise JsonLdError(
+                    'Could not dereference URL; can only load URLs using',
+                    'the "http" and "https" schemes.',
+                    'jsonld.InvalidUrl', {'url': url},
+                    code='loading document failed')
+
+            response = self.session.get(
+                url, headers={'Accept': 'application/ld+json, application/json'})
+
+            doc = {'contextUrl': None, 'documentUrl': url, 'document': response.text}
+
+            if self.cachable:
+                doc['from_cache'] = response.from_cache
+                self.session.remove_expired_responses()
```

Comment:
Note that, despite the requests_cache docs, removed_expired_responses is a method on the `CachedSession` itself and not (it seems) on `CachedSession.cache`. Their API docs places it on the CachedSession directly, but their user docs say it's on `CachedSession.cache`. 8P